browser/net/brave_localhost_permission_network_delegate_helper.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/net/brave_localhost_permission_network_delegate_helper.h"
+
+#include <array>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "brave/browser/brave_browser_process.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/permission_controller_delegate.h"
+#include "content/public/browser/web_contents.h"
+#include "net/base/net_errors.h"
+#include "net/base/url_util.h"
+#include "third_party/blink/public/common/permissions/permission_utils.h"
+
+namespace brave {
+
+void OnPermissionRequestStatus(
+    content::WebContents* contents,
+    const std::vector<blink::mojom::PermissionStatus>& permission_statuses) {
+  DCHECK_EQ(1u, permission_statuses.size());
+  if (contents &&
+      permission_statuses[0] == blink::mojom::PermissionStatus::GRANTED) {
+    contents->GetController().Reload(content::ReloadType::NORMAL, true);
```

Comment:
The issue we were running into was that users would get an error message from the page because we blocked the request, and then it wouldn't go away, so they wouldn't know to try the action again. Reloading the tab gets them to know that they took an action i.e. granted a permission, now they should try again.

Reply:
i wonder how relevant that would be, if the page already navigated away from the original

Reply:
Can you say more? This just seemed like the safest thing to do, because once you've reloaded the page will try to redo whatever it was trying before it was throttled, so there's no question of weird error states. Similarly for users, they would know to redo whatever they were trying to do instead of being confused by an error message.